src/ray/object_manager/plasma/client.cc

```diff
@@ -751,8 +756,14 @@ Status PlasmaClient::Impl::EnsureGetAcquired(
   }
 
   int64_t version_read = 0;
+
+  // Need to unlock the client mutex since ReadAcquire() is blocking.
+  // TODO(ekl) is this entirely thread-safe?
```

Comment:
Looks like the header is not thread safe. This may cause issues when there are 2 client calls from 2 different thread accessing the same header?